File: mario.py
# import libraries
from cs50 import get_int

# get user input
# only allow integers between 1-8 inclusive
while True:
    height = get_int("Height: ")
    if height >= 1 and height <= 8:
        break

# print pyramid of inputed height
j = 1
for i in range(height):
    for k in range(height - j):
        print(" ", end="")
    for n in range(j):
        print("#", end="")
    print("")
    j = j + 1

File: mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // declares the variable "height"
    int height;

    // prompts the user to input a number between 1 and 8
    // the user is reprompted if the input any other number
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // declares variables i, j, and k
    int i;
    int j;
    int k;

    // prints a pyramid to the screen that has a height and width of the inputed number
    for (i = 0; i < height; i++)
    {
        // prints a certain number of spaces on each line so the pyramid is configured towards the right-hand side
        for (j = height - (i + 2); j >= 0; j--)
        {
            printf(" ");
        }

        // prints the "blocks" of the pyramid for each line
        for (k = 1; k <= (i + 1); k++)
        {
            printf("#");
        }
        printf("\n");
    }
}